src/frontend/src/handler/describe.rs

```diff
@@ -52,35 +35,61 @@ pub async fn handle_describe(
     let catalog_reader = session.env().catalog_reader().read_guard();
 
     // For Source, it doesn't have table catalog so use get source to get column descs.
-    let columns: Vec<ColumnDesc> = {
-        let catalogs = match catalog_reader
+    let (columns, indexs): (Vec<ColumnDesc>, Vec<TableCatalog>) = {
+        let (catalogs, indexs) = match catalog_reader
             .get_schema_by_name(session.database(), &schema_name)?
             .get_table_by_name(&table_name)
         {
-            Some(table) => &table.columns,
-            None => {
+            Some(table) => (
+                &table.columns,
+                catalog_reader
+                    .get_schema_by_name(session.database(), &schema_name)?
+                    .iter_index()
+                    .filter(|x| x.is_index_on == Some(table.id))
+                    .cloned()
+                    .collect_vec(),
+            ),
+            None => (
                 &catalog_reader
                     .get_source_by_name(session.database(), &schema_name, &table_name)?
-                    .columns
-            }
+                    .columns,
+                vec![],
```

Comment:
Maybe leave a TODO here and create an issue. I think source (non materialized) can also have indices in the future. https://github.com/singularity-data/risingwave/issues/2738